recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,4 @@
+MACOSX_SDK_VERSION:        # [osx and x86_64]
+  - "10.14"                # [osx and x86_64]
+MACOSX_DEPLOYMENT_TARGET:  # [osx and x86_64]
+  - "10.14"                # [osx and x86_64]
```

Comment:
Adding this to match minimal deployment target supported by rocksdb, and mirroring what [rocksdb-feedstock](https://github.com/conda-forge/rocksdb-feedstock/blob/9759b498504868af606bcc9ebbe09d10f1b2cff0/recipe/conda_build_config.yaml) does